widgetssdk/src/main/java/com/glia/widgets/view/unifiedui/UnifiedUiExtensions.kt

```diff
@@ -51,58 +51,71 @@ internal fun createBackgroundFromTheme(color: ColorTheme): Drawable = color.run
     }
 }
 
+internal val View.backgroundGradientDrawable: GradientDrawable
+    get() = when {
+        // The part below is needed to preserve GradientDrawable/background properties that are `null`.
+        // For example without this implementation if `cornerRadius = null` but `stroke = Color.RED` then
+        // whatever corner radios is in the default UI implementation it would automatically become 0 instead
+        background is StateListDrawable && background.current is StateListDrawable -> background.current as GradientDrawable
+        background is GradientDrawable -> background as GradientDrawable
+        else -> GradientDrawable()
+    }
+
 /**
  * will update whole background without keeping an old background or stroke
  * in case when the old background differs from [GradientDrawable]
  */
-internal fun View.applyLayerTheme(defTheme: LayerTheme?, disabledTheme: LayerTheme? = null) {
-    if (defTheme?.fill == null && defTheme?.stroke == null) return
+internal fun View.applyLayerTheme(layerTheme: LayerTheme?) {
```

Comment:
This change resulted in the 'snapshots' modifications, as we don't need `StateListDrawable' for views that have only one default state.

Reply:
Isn't it better to replace snapshot files only for the cases that contain changes to avoid missing some undesired changes? Or you mean that it's not possible in this PR?